File: AOJ/0396/treasure_map.java
import java.util.Scanner;

class Main {
  void solve(){
    Scanner sc = new Scanner( System.in );
    int w = sc.nextInt();
    int h = sc.nextInt();

    Integer[] col = new Integer[w];
    Integer[] row = new Integer[h];
    int sumC = 0, sumR = 0;
    for ( int i = 0; i < w; ++i ) {
      col[i] = sc.nextInt();
      sumC += col[i];
    }
    for ( int i = 0; i < h; ++i ) {
      row[i] = sc.nextInt();
      sumR += row[i];
    }
    if ( sumR != sumC ) {
      System.out.println("0");
      return;
    }

    for ( int i = 0; i < w; ++i ) {

      Arrays.sort(row, Comparator.reverseOrder()); //降順ソート
	    
      for ( int j = 0; j < h; ++j ) {
        if ( col[i] == 0 || row[j] == 0 ) break;
        --row[j];
        --col[i];
      }

      if ( col[i] > 0 ) {
        System.out.println("0");
        return;
      }
    }

    System.out.println("1");
  }

  public static void main(String[] args) { new Main().solve(); }
}

File: AOJ/0396/treasure_map.py
w,h = map(int, input().split())
sumC = 0
sumR = 0
col = list(map(int, input().split()))
row = list(map(int, input().split()))
for c in col : sumC += c
for r in row : sumR += r

if sumR != sumC :
    print(0)
    sys.exit(0)

for i in range(w):
    row.sort(reverse = True)
    for j in range(h):
        if not col[i] or not row[j] : break
        row[j] -= 1
        col[i] -= 1
    if col[i] > 0 :
        print(0)
        sys.exit(0)
print(1)
